### PR TITLE
Add recipe for gr-bladeRF

### PR DESCRIPTION
**Pull request by rghilduta**

### CODE REVIEW

File: gr-bladeRF.lwr [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+#
+# You should have received a copy of the GNU General Public License
+# along with PyBOMBS; see the file COPYING.  If not, write to
+# the Free Software Foundation, Inc., 51 Franklin Street,
+# Boston, MA 02110-1301, USA.
+#
+
+category: common
+depends:
+- uhd
```
Comment (willcode) [inline gr-bladeRF.lwr:22]: Does `gr-bladeRF` really depend on all these packages?

**Review by willcode — COMMENTED** [review]
Check dependencies and config opts.

File: gr-bladeRF.lwr (hunk 1 above)
Comment (rghilduta) [reply gr-bladeRF.lwr]: Nope it does not, updated!

File: gr-bladeRF.lwr [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+depends:
+- gnuradio
+- bladeRF
+description: Interface API independent of the underlying radio hardware
+gitbranch: main
+inherit: cmake
+# Let's always build this from source, not binaries
+source: git+https://github.com/Nuand/gr-bladeRF.git
+vars:
+  config_opt: ' -DENABLE_NONFREE=TRUE '
```
[the remark this answers is not in the document]
Comment (Nuand) [reply gr-bladeRF.lwr]: Fixed and updated a bunch of other things (description).

**Review by willcode — APPROVED** [review]

**Comment by argilo** [comment]
This builds fine with GNU Radio 3.9, so I'll merge it. You might want to look into build failures with GNU Radio 3.10 and main.

GNU Radio 3.10:
```
[  6%] Building CXX object lib/CMakeFiles/gnuradio-bladeRF.dir/source_impl.cc.o
In file included from /home/argilo/prefix_310/src/gr-bladeRF/lib/source_impl.cc:8:
/home/argilo/prefix_310/src/gr-bladeRF/lib/source_impl.h:12:10: fatal error: bladerf_source_c.h: No such file or directory
   12 | #include "bladerf_source_c.h"
      |          ^~~~~~~~~~~~~~~~~~~~
compilation terminated.
make[2]: *** [lib/CMakeFiles/gnuradio-bladeRF.dir/build.make:76: lib/CMakeFiles/gnuradio-bladeRF.dir/source_impl.cc.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:251: lib/CMakeFiles/gnuradio-bladeRF.dir/all] Error 2
make: *** [Makefile:146: all] Error 2
```

GNU Radio main branch:
```
[  5%] Building CXX object lib/CMakeFiles/gnuradio-bladeRF.dir/source_impl.cc.o
In file included from /home/argilo/prefix_311/src/gr-bladeRF/lib/bladerf/common_sync_block.h:3,
                 from /home/argilo/prefix_311/src/gr-bladeRF/lib/bladerf/bladerf_source_c.h:24,
                 from /home/argilo/prefix_311/src/gr-bladeRF/lib/source_impl.h:12,
                 from /home/argilo/prefix_311/src/gr-bladeRF/lib/source_impl.cc:8:
/home/argilo/prefix_311/src/gr-bladeRF/lib/bladerf/bladerf_common.h: In member function ‘T bladerf_common::get_pmic(bladerf_pmic_register)’:
/home/argilo/prefix_311/src/gr-bladeRF/lib/bladerf/bladerf_common.h:73:26: error: ‘str’ is not a member of ‘boost’
   73 |     BLADERF_THROW(boost::str(boost::format("%s: %s (%d)") % message         \
      |                          ^~~
/home/argilo/prefix_311/src/gr-bladeRF/lib/bladerf/bladerf_common.h:68:65: note: in definition of macro ‘BLADERF_THROW’
   68 | throw std::runtime_error(std::string(__FUNCTION__) + ": " + message);   \
      |                                                             ^~~~~~~

/home/argilo/prefix_311/src/gr-bladeRF/lib/bladerf/bladerf_common.h:186:9: note: in expansion of macro ‘BLADERF_THROW_STATUS’
  186 |         BLADERF_THROW_STATUS(status, "bladerf_get_pmic_register failed");
      |         ^~~~~~~~~~~~~~~~~~~~
/home/argilo/prefix_311/src/gr-bladeRF/lib/bladerf/bladerf_common.h:73:37: error: ‘format’ is not a member of ‘boost’; did you mean ‘fmt::v8::format’?
   73 |     BLADERF_THROW(boost::str(boost::format("%s: %s (%d)") % message         \
      |                                     ^~~~~~
/home/argilo/prefix_311/src/gr-bladeRF/lib/bladerf/bladerf_common.h:68:65: note: in definition of macro ‘BLADERF_THROW’
   68 | throw std::runtime_error(std::string(__FUNCTION__) + ": " + message);   \
      |                                                             ^~~~~~~

/home/argilo/prefix_311/src/gr-bladeRF/lib/bladerf/bladerf_common.h:186:9: note: in expansion of macro ‘BLADERF_THROW_STATUS’
  186 |         BLADERF_THROW_STATUS(status, "bladerf_get_pmic_register failed");
      |         ^~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/spdlog/fmt/fmt.h:26,
                 from /usr/include/spdlog/common.h:36,
                 from /home/argilo/prefix_311/include/gnuradio/logger.h:36,
                 from /home/argilo/prefix_311/include/gnuradio/custom_lock.h:15,
                 from /home/argilo/prefix_311/include/gnuradio/buffer.h:15,
                 from /home/argilo/prefix_311/include/gnuradio/buffer_double_mapped.h:15,
                 from /home/argilo/prefix_311/include/gnuradio/io_signature.h:23,
                 from /home/argilo/prefix_311/include/gnuradio/basic_block.h:15,
                 from /home/argilo/prefix_311/include/gnuradio/hier_block2.h:15,
                 from /home/argilo/prefix_311/src/gr-bladeRF/lib/../include/bladeRF/source.h:12,
                 from /home/argilo/prefix_311/src/gr-bladeRF/lib/source_impl.h:11,
                 from /home/argilo/prefix_311/src/gr-bladeRF/lib/source_impl.cc:8:
/usr/include/fmt/format.h:3058:13: note: ‘fmt::v8::format’ declared here
 3058 | inline auto format(const Locale& loc, format_string<T...> fmt, T&&... args)
      |             ^~~~~~
make[2]: *** [lib/CMakeFiles/gnuradio-bladeRF.dir/build.make:76: lib/CMakeFiles/gnuradio-bladeRF.dir/source_impl.cc.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:268: lib/CMakeFiles/gnuradio-bladeRF.dir/all] Error 2
make: *** [Makefile:146: all] Error 2
```

**Review by argilo — APPROVED** [review]

**Comment by argilo** [comment]
The failures on the main branch look to be due to `bladerf_common.h` using `boost::str` without including `<boost/format.hpp>`.

**Comment by argilo** [comment]
The failure with 3.10 looks to be due to `lib/source_impl.h` not having the correct relative path to `bladerf_source_c.h`. Changing it to `#include "bladerf/bladerf_source_c.h"` solved that problem.

**Pull request merged**